test/test_images/sender/main.go

```diff
@@ -45,11 +69,13 @@ func main() {
 	ctx, span := trace.StartSpan(context.Background(), "sender", trace.WithSampler(trace.AlwaysSample()))
 	defer span.End()
 
-	ctx, _, err = ceClient.Send(ctx, dummyCloudEvent())
-	rtctx := cloudevents.HTTPTransportContextFrom(ctx)
+	// If needRetry is true, repeat sending Event with exponential backoff when there is some specific errors.
+	// 404 error indicates some broker configmap sync up issue and 5XX error indicates credential sync issues.
```

Comment:
To be precise, 404 could also mean the broker was never created and 5xx could mean our code has bug. So better to document the other way around - like sync problems could cause 404 and 5xx and in e2e tests retrying those would help reduce flakiness.